backend/logic/curriculums/curriculums.js

```diff
@@ -0,0 +1,95 @@
+const mysql = require("../../sql/connection");
+const format = require("../../validation/format");
+
+var createCurriculum = async (name, type, department, numOfElectives, cores, techComps, comps) => {
+    let error = false;
+  
+    if (!format.verifyCurriculumName(name)) {
+      error = "Invalid curriculum name";
+    } else if(!format.verifyCurrType(type)){
+      error = "Invalid curriculum type";
+    } else if(!format.verifyDepartment(department)){
+      error = "Invalid department";
+    } else if(!format.verifyNumOfElectives(numOfElectives)) {
+      error = "Invalid number of electives"
+    }
+  
+    if (!error == false) {
+      console.error(error);
+      throw new Error(error);
+    }
+  
+    format.verifyCourse(cores);
```

Comment:
Shouldn't you return an error if upon verification, it turns out to be invalid?

Reply:
good verification, follow up to feras's comment, if it throws an error, how will you handle it? Could you potentially put it within the try catch below? This way you lockdown any errors that happen

Reply:
This function is called by routes, where the function is surrounded by a try catch and ends back a 500 if an error was detected.

Reply:
Ok sorry, didn't realize that verifyCourses() threw an error and didnt just return a boolean like the other format validation functions.